newIDE/app/src/GameDashboard/Monetization/UserEarningsWidget.js

```diff
@@ -0,0 +1,252 @@
+import CreditOutDialog from './CashOutDialog';
+import GDevelopThemeContext from '../../UI/Theme/GDevelopThemeContext';
+import DashboardWidget from '../Widgets/DashboardWidget';
+import { useResponsiveWindowSize } from '../../UI/Responsive/ResponsiveWindowMeasurer';
+
+const styles = {
+  separator: {
+    height: 50,
+  },
+  buttonContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'stretch',
+  },
+};
+
+type Props = {|
+  fullWidth?: boolean,
+|};
+
+const UserEarningsWidget = ({ fullWidth }: Props) => {
+  const {
+    userEarningsBalance,
+    onRefreshEarningsBalance,
+    onRefreshLimits,
+  } = React.useContext(AuthenticatedUserContext);
+  const theme = React.useContext(GDevelopThemeContext);
+  const { isMobile } = useResponsiveWindowSize();
+
+  const [earningsInMilliUsd, setEarningsInMilliUsd] = React.useState(0);
+  const [earningsInCredits, setEarningsInCredits] = React.useState(0);
+  const [error, setError] = React.useState(null);
+  const intervalValuesUpdate = React.useRef(null);
+
+  const [selectedCashOutType, setSelectedCashOutType] = React.useState<
+    ?'cash' | 'credits'
+  >(null);
+
+  const fetchUserEarningsBalance = React.useCallback(
+    async () => {
+      if (!userEarningsBalance) return;
+
+      try {
+        // Create an animation to show the earnings increasing.
+        const targetMilliUsd = userEarningsBalance.amountInMilliUSDs;
+        const targetCredits = userEarningsBalance.amountInCredits;
+
+        const duration = 500;
+        const steps = 30;
+        const intervalTime = duration / steps;
+
+        const milliUsdIncrement = (targetMilliUsd - earningsInMilliUsd) / steps;
+        const creditsIncrement = (targetCredits - earningsInCredits) / steps;
+
+        let currentMilliUsd = earningsInMilliUsd;
+        let currentCredits = earningsInCredits;
+        let step = 0;
+
+        intervalValuesUpdate.current = setInterval(() => {
+          step++;
+          currentMilliUsd += milliUsdIncrement;
+          currentCredits += creditsIncrement;
+
+          setEarningsInMilliUsd(currentMilliUsd);
+          setEarningsInCredits(currentCredits);
+
+          if (step >= steps) {
+            clearInterval(intervalValuesUpdate.current);
+            // Ensure final values are exactly the target values
+            setEarningsInMilliUsd(targetMilliUsd);
+            setEarningsInCredits(targetCredits);
+          }
+        }, intervalTime);
+      } catch (error) {
+        console.error('Unable to get user earnings balance:', error);
+        setError(error);
+      }
+    },
+    [userEarningsBalance, earningsInMilliUsd, earningsInCredits]
+  );
+
+  React.useEffect(
+    () => {
+      fetchUserEarningsBalance();
+    },
+    // Fetch the earnings once on mount.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  React.useEffect(
+    () => () => {
+      // Cleanup the interval when the component is unmounted.
```

Comment:
That's clean!
Just for thought: without the comment, I would have believed this effect is run on mount because the `() => () => {` is hard to notice.
Do you think it would be more legible with `() => { return () => {`?

Reply:
Unsure! I think in both cases, the comment is needed